runtime/src/main/java/com/couchbase/quarkus/extension/runtime/health/CouchbaseReadyCheck.java

```diff
@@ -0,0 +1,111 @@
+import org.eclipse.microprofile.health.Readiness;
+
+import com.couchbase.client.core.Core;
+import com.couchbase.client.core.CoreContext;
+import com.couchbase.client.core.Reactor;
+import com.couchbase.client.core.config.ConfigVersion;
+import com.couchbase.client.core.msg.ResponseStatus;
+import com.couchbase.client.core.msg.kv.CarrierGlobalConfigRequest;
+import com.couchbase.client.core.retry.BestEffortRetryStrategy;
+import com.couchbase.client.core.topology.ClusterTopology;
+import com.couchbase.client.core.topology.NodeIdentifier;
+import com.couchbase.client.java.Cluster;
+import com.couchbase.quarkus.extension.runtime.CouchbaseConfig;
+
+import io.smallrye.health.api.AsyncHealthCheck;
+import io.smallrye.mutiny.Uni;
+import io.smallrye.mutiny.converters.uni.UniReactorConverters;
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+
+@Readiness
+@ApplicationScoped
+public class CouchbaseReadyCheck implements AsyncHealthCheck {
+
+    @Inject
+    Cluster cluster;
+
+    @Inject
+    CouchbaseConfig config;
+
+    /**
+     * Wait for a Cluster config, then pings the GCCCP connections on each node.
+     *
+     * @return A HealthCheckResponse.
+     */
+    @Override
+    public Uni<HealthCheckResponse> call() {
+        HealthCheckResponseBuilder builder = HealthCheckResponse.named("Couchbase Cluster Readiness Check");
+
+        return Uni.createFrom()
+                .converter(UniReactorConverters.fromMono(), quarkusWaitUntilReady(cluster))
+                .map(ignored -> builder.up().build())
+                .onFailure().recoverWithItem(throwable -> builder.down()
+                        .withData("error", throwable.getMessage())
+                        .build());
+    }
+
+    public Mono<Void> quarkusWaitUntilReady(Cluster cluster) {
+        return cluster.reactive()
+                .waitUntilReady(
+                        Duration.ofSeconds(config.readinessTimeout()),
+                        waitUntilReadyOptions().serviceTypes(KV) // Can't really wait for KV -- this just waits for global config
+                )
+                .then(Mono.defer(() -> {
+                    // Make sure every KV node in the global config is still online
+                    ClusterTopology clusterTopology = requireNonNull(
+                            cluster.core().configurationProvider().config().globalTopology(), "Cluster topology isn't loaded?");
+                    return Flux.fromIterable(clusterTopology.nodes())
+                            .filter(it -> it.has(KV))
+                            .flatMap(node -> pingGcccp(cluster.core(), node.id()))
+                            .then();
+                }));
+    }
+
+    private Mono<Void> pingGcccp(Core core, NodeIdentifier nodeId) {
+        final CoreContext ctx = core.context();
+        return Mono.defer(() -> {
+            CarrierGlobalConfigRequest request = new CarrierGlobalConfigRequest(
```

Comment:
There is a NOOP memcached request that would technically be better than a rather heavy GET_CONFIG.  But we haven't exposed support for it, so this will do for now - on the understanding that the health-check is generally configured to run every X seconds, not millis.